yt/frontends/nc4_cm1/data_structures.py

```diff
@@ -0,0 +1,233 @@
+                dim_info[dim] = meta.size
+            self.parameters["dimensions"] = dim_info
+
+        self.parameters["coords"] = coords
+
+        self.dimensionality = 3
+        self.domain_dimensions = np.array(dims, dtype="int64")
+        self.periodicity = (False, False, False)
+        self.parameters["time"] = self.current_time
+
+        # Set cosmological information to zero for non-cosmological.
+        self.cosmological_simulation = 0.0
+        self.current_redshift = 0.0
+        self.omega_lambda = 0.0
+        self.omega_matter = 0.0
+        self.hubble_constant = 0.0
+
+    @classmethod
+    def _is_valid(self, *args, **kwargs):
+        # This accepts a filename or a set of arguments and returns True or
+        # False depending on if the file is of the type requested.
+
+        warn_netcdf(args[0])
+        try:
+            from netCDF4 import Dataset
+
+            with Dataset(args[0], "r", keepweakref=True) as ds:
+                is_cm1_lofs = hasattr(ds, "cm1_lofs_version")
+                is_cm1 = hasattr(ds, "cm1 version")
+                # ensure coordinates of each variable array exists in the dataset
+                coords = ds.dimensions  # get the dataset wide coordinates
+                failed_vars = []  # list of failed variables
+                for var in ds.variables:  # iterate over the variables
+                    vcoords = ds[var].dimensions  # get the dimensions for the variable
+                    ncoords = len(vcoords)  # number of coordinates in variable
+                    coordspassed = 0  # number of coordinates that pass for a variable
+                    for vc in vcoords:  # iterate over the coordinates for the variable
+                        if vc in coords:
+                            # variable coordinate and global coordinate are same
+                            coordspassed += 1
+                    if coordspassed != ncoords:
+                        failed_vars.append(var)
+
+                if len(failed_vars) > 0:
+                    mylog.error(
+                        (
+                            "Trying to load a cm1_lofs netcdf file but the coordinates "
+                            "of the following fields do not match the coordinates of "
+                            f"the dataset: {failed_vars}"
+                        )
+                    )
+                    return False
+
+            if is_cm1_lofs is False:
+                if is_cm1:
+                    mylog.error(
+                        (
+                            "It looks like you are trying to load a cm1 netcdf file, "
+                            "but at present yt only supports cm1_lofs output. Until"
+                            " support is added, you can likely use"
+                            " yt.load_uniform_grid() to load your cm1 file manually."
+                        )
+                    )
+                return False
+        except Exception:
```

Comment:
Exception catching should be as specific as possible, can you please catch only the relevant exceptions here ?